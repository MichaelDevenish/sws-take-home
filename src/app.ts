
import express from 'express'
import gql from "graphql-tag";
import {ApolloServer} from "@apollo/server";
import {expressMiddleware} from "@apollo/server/express4";
import bodyParser from "body-parser";
import cors from 'cors'
import {typeDefs, resolvers} from "./domains";

const app = express()

const server = new ApolloServer({ typeDefs, resolvers });

app.use(cors())
app.use(bodyParser.json())

async function startServer() {
    await server.start()
    app.use('/graphql', expressMiddleware(server))
}
startServer();

app.listen(3000, function () {
    console.log('graphql app listening on port 3000!');
});

export {
    app
}
