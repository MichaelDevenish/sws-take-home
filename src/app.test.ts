import {app} from "./app";
import request from "supertest";

describe("Test the graphql api", () => {
    test("It should complain if no query string is provided", async () => {
        const response = await request(app)
            .post("/graphql")
            .set('Accept', 'application/json')
            .set('Content-Type', 'application/json')

        expect(response.body).toMatchSnapshot()
        expect(response.status).toEqual(400)
    });
});