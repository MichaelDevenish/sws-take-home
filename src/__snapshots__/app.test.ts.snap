// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the graphql api It should complain if no query string is provided 1`] = `
{
  "errors": [
    {
      "extensions": {
        "code": "BAD_REQUEST",
      },
      "message": "POST body missing, invalid Content-Type, or JSON object has no keys.",
    },
  ],
}
`;
